treeherder/webapp/api/performance_data.py

```diff
@@ -882,6 +904,84 @@ def _get_signatures(self, repository_name, framework, interval, no_subtests):
         signatures = self._get_signatures_values(signatures)
         return signatures
 
+    @staticmethod
+    def _get_ph_time_ranges():
+        return [
+            {'value': 86400, 'text': 'Last day'},
+            {'value': 86400 * 2, 'text': 'Last 2 days'},
+            {'value': 604800, 'text': 'Last 7 days'},
+            {'value': 1209600, 'text': 'Last 14 days'},
+            {'value': 2592000, 'text': 'Last 30 days'},
+            {'value': 5184000, 'text': 'Last 60 days'},
+            {'value': 7776000, 'text': 'Last 90 days'},
+            {'value': 31536000, 'text': 'Last year'},
+        ]
+
+    @staticmethod
+    def _get_graphs_link(seriesList, resultSets, timeRange):
+        params = {
+            'series': map(
+                lambda series: [
+                    series['projectName'],
+                    series['signature'],
+                    1,
+                    series['frameworkId'],
+                ],
+                seriesList,
+            ),
+            'highlightedRevisions': map(lambda resultSet: resultSet['revision'][0:12], resultSets),
+        }
+
+        if resultSets and not timeRange:
+            timeRanges = map(
+                lambda phTimeRanges: {'range': phTimeRanges['value']},
+                PerformanceSummary._get_ph_time_ranges(),
+            )
+
+            params['timerange'] = max(
+                map(
+                    lambda resultSet: filter(
+                        lambda ranges: (datetime.now() / 1000.0) - resultSet['push_timestamp']
+                        < ranges,
+                        timeRanges,
+                    ),
+                    resultSets,
+                )
+            )
+
+        if timeRange:
+            params['timerange'] = timeRange
+
+        return './graphs?%s' % urlencode(params)
```

Comment:
@beatrice-acasandrei I have series, timeRange, highlightedRevisions here
I'm not sure if highlightedrvisions for new and old are merged together, but I'll check it out.

Also, I'll look into adding the signature_hash